### PR TITLE
[CLEANUP] Cleanup IE11 leftovers

### PR DESCRIPTION
**Pull request by simonihmig**
Further cleanup after dropping IE11 support: (best reviewed commit after commit)
* Use WeakSet for lazy events
* Remove Object.entries/.values polyfills
* Remove HAS_NATIVE_SYMBOL and HAS_NATIVE_PROXY flags (as all supported browser support these)

### CODE REVIEW

**Comment by nlfurniss** [comment]
@simonihmig small correction: `Remove Object.assign/.value polyfills` --> `Remove Object.entries/.values polyfills`

**Comment by simonihmig** [comment]
Oh, right, corrected. Thanks!

File: packages/@ember/-internals/routing/lib/services/router.ts [hunk 1]
```diff
@@ -10,7 +10,7 @@ import Route from '../system/route';
 import EmberRouter, { QueryParam } from '../system/router';
 import { extractRouteArgs, resemblesURL, shallowEqual } from '../utils';
 
-const ROUTER = symbol('ROUTER') as string;
+const ROUTER = (symbol('ROUTER') as unknown) as string;
```
Comment (mixonic) [inline packages/@ember/-internals/routing/lib/services/router.ts:13]: What is the motivation for this change?

File: yarn.lock [hunk 2]
```diff
@@ -955,6 +955,8 @@
   version "7.12.1"
   resolved "https://registry.yarnpkg.com/@babel/plugin-transform-for-of/-/plugin-transform-for-of-7.12.1.tgz#07640f28867ed16f9511c99c888291f560921cfa"
   integrity sha512-Zaeq10naAsuHo7heQvyV0ptj4dlZJwZgNAtBYBnu5nNKJoW62m0zKcIEyVECrUKErkUkg6ajMy4ZfnVZciSBhg==
+  dependencies:
```
Comment (mixonic) [inline yarn.lock:958]: Should there be `yarn.lock` changes in this PR? I think not?

**Review by mixonic — CHANGES_REQUESTED** [review]
Very cool. This might have been easier as a distinct PR for each change, but I'm mostly comfortable with it after scrolling through it.

Left two notes to get clarity on.

**Comment by rwjblue** [comment]
Bah, I thought this would just update to point at master :(

**Comment by mixonic** [comment]
@simonihmig this PR was auto-closed when `v4-cleanup` was merged to `master`. Can you re-open for us to review/merge against `master`? This is excellent work and I don't want to lose it.

File: yarn.lock (hunk 2 above)
Comment (simonihmig) [reply yarn.lock:958]: Probably somehow by accident, removed that in my reopened PR!
